concordium-std/src/lib.rs

```diff
@@ -273,9 +277,19 @@ pub use impls::*;
 pub use traits::*;
 pub use types::*;
 
+#[cfg(feature = "dl-malloc")]
+extern crate dlmalloc;
+// Use `globalDlmalloc` as the global allocator.
+#[allow(dead_code)]
+#[cfg(feature = "dl-malloc")]
+#[cfg_attr(not(feature = "std"), global_allocator)]
+static ALLOC: dlmalloc::GlobalDlmalloc = dlmalloc::GlobalDlmalloc;
```

Comment:
The idea (as I thought it) was to just have the `wee-alloc` feature. 

And we document how users can use their own allocator if they want. But we would not be supporting multiple allocators in concordium-std.